internal/core/command/device.go

```diff
@@ -37,7 +38,9 @@ func executeCommandByDeviceID(
 	isPutCommand bool,
 	lc logger.LoggingClient,
 	dbClient interfaces.DBClient,
-	deviceClient metadata.DeviceClient) (string, error) {
+	deviceClient metadata.DeviceClient,
+	r *http.Request,
+	httpCaller internal.HttpCaller) (string, error) {
```

Comment:
These arguments can be reduced since `http.Request` is now being passed in and some of the other arguments can be obtained from the request. Please apply this comment to the other functions down the call stack.